[org] Improvements when using org-modern-mode

bcc32 opened the pull request:
When using org-modern-mode:

- Disable org-superstar from the spacemacs-org layer.  These packages
  have overlapping functionality, and org-modern's README recommends
  not using org-superstar at the same time.

- Use `global-org-modern-mode` directly rather than reimplementing it.
  This allows it to be easily disabled globally

- Add a key binding to toggle `global-org-modern-mode`

bcc32 committed:
configuration-layer/package-used-p: Check package :toggle non-nil

This makes configuration-layer/package-used-p behave the same as
cfgl-package-used-p for the same package, which should have been the
case all along.

bcc32 committed:
cfgl-package-reqs-satisfied-p: Check if required package is used

A package's dependencies are only satisfied if the dependency is
actually used (:toggle evaluates to non-nil).  If the dependency is
:toggled off, then the dependent package will not be installed or
used, so this function should reflect that.

bcc32 committed:
[spacemacs-org] Disable org-superstar when org-modern is enabled

org-modern provides similar functionality for the leading stars in org
headlines and it is recommended to disable org-superstar when
org-modern is used.

fnussbaum commented on layers/+spacemacs/spacemacs-org/packages.el:
```diff
@@ -33,7 +33,7 @@
     ;; `org' package.
     (default-org-config :location built-in)
-    org-superstar
+    (org-superstar :toggle (not (configuration-layer/package-used-p 'org-modern)))
```
This seems to depend on a solution for #16346 which is not implemented or pushed yet?

bcc32 replied:
Oh, sad.  It works on my branch because I have a WIP fix for #16346.  Thanks for catching this.  Making the PR into a draft for now.

bcc32 committed:
[layouts] Remove workaround for configuration-layer/package-used-p

Now that configuration-layer/package-used-p checks the package :toggle
form, there is no need to also check the mode line theme name.

bcc32 committed:
WIP lots of code review
